docs/vNext/Benchmark_Generator.md

```diff
@@ -0,0 +1,105 @@
+
+```console
+Welcome to the Hyperledger Caliper benchmark generator!
+Let's start off by creating a workspace folder!
+? What would you like to call your workspace? myBenchmark
+```
+
+### Callback Prompts
+The generator will inititally take you through generating the callback file and you will be prompted for:
+* a **callback**
+* the contract ID, which is the **name** of the smart contract
+* the smart contract **version**
+
+By the end, you should have something similar to the following:
+
+```console
+Now for the callback file...
+? What would you like to name you callback file? (this could be the name of the operation you are testing) createCar
+? What is the ID of your contract? Fabcar
+? What is the contract version? 1.0.0
+```
+
+### Configuration Prompts
+Next, you will be taken through generating the [configuration file](./Architecture#ConfigurationFile) and you will be prompted for:
+* the **name** of the benchmark
+* a **description** of the benchmark
+* the number of **clients**
+> __Note:__ On an invalid input value for clients, a default value will be used.
+* a **label** for differentiating between multiple rounds
+* the **rate controller** you would like to use. The generator currently provides the rate controllers displayed below as options. The generated configuration file will use default `opts` for whichever rate controller is chosen.
+
+```console
+? Rate controller: (Use arrow keys)
+❯ Fixed Rate 
+  Fixed Backlog 
+  Linear Rate 
+  Fixed Feedback Rate
+  ```
+* the method of which you should like to measure the length of the round. The round may be measured using either `txDuration`, which defines the length of the round in seconds, or `txNumber`, which defines the length of the round using the number of transactions to be generated in the round.
+
+```console
+? How would you like to measure the length of the round? (Use arrow keys)
+❯ Transaction Duration 
+  Transaction Number
+```
+* a value for either **txNumber** or **txDuration** depending on the answer to previous prompt.
+
+> __Note:__ On an invalid input value for either txDuration or txNumber, a default value will be used
+
+By the end, you should have something similar to the following:
+
+```console
+Now for the benchmark configuration file...
+? Name of benchmark: Fabcar benchmark
+? Description of benchmark: Benchmark for performance testing fabcar contract modules
+? Number of clients: 5
+? Label: (hint for the test) Round for creating a car 
+? Rate controller: Fixed Rate
+? How would you like to measure the length of the round? Transaction Number
+? How many transactions would you like to have in this round? 60
+```
+
+On successful generation, you should see the following:
+
+```console
+Generating benchmark configuration file...
+   create myBenchmark/benchmarks/callbacks/createCar.js
+   create myBenchmark/benchmarks/config.yaml
+Finished generating benchmark configuration file
+```
```

Comment:
It would be nice to discuss what should be generated as a last part, for instance a more wordy version of "you get the files placed within the workspace directory". I suspect you might need to add a comment on what a user will still need to provide themselves (ie a network file) ... possibly also a note on other aspects a user might extend from